packages/batch-submitter/src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -742,67 +689,27 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
     this.log.debug('Fetched L2 block', {
       block,
     })
-    const txType = block.transactions[0].txType
+
+    const batchElement = {
+      stateRoot: block.stateRoot,
+      timestamp: block.timestamp,
+      blockNumber: block.transactions[0].l1BlockNumber,
+      isSequencerTx: undefined,
+      rawTransaction: undefined,
+    }
 
     if (this._isSequencerTx(block)) {
-      if (txType === TxType.EIP155 || txType === TxType.EthSign) {
-        return this._getDefaultEcdsaTxBatchElement(block)
-      } else {
-        throw new Error('Unsupported Tx Type!')
-      }
-    } else {
-      return {
-        stateRoot: block.stateRoot,
-        isSequencerTx: false,
-        sequencerTxType: undefined,
-        txData: undefined,
-        timestamp: block.timestamp,
-        blockNumber: block.transactions[0].l1BlockNumber,
-      }
+      batchElement.isSequencerTx = true
+      batchElement.rawTransaction = block.transactions[0].rawTransaction
     }
+
+    return batchElement
   }
 
   private async _getBlock(blockNumber: number): Promise<L2Block> {
-    const block = (await this.l2Provider.getBlockWithTransactions(
+    return (await this.l2Provider.getBlockWithTransactions(
```

Comment:
The `await` here is redundant unless its required for typescript casting but I feel like you could do

```ts
return this.l2Provider.getBlockWithTransactions(blockNumber) as Promise<L2Block>
```

Reply:
Btw had to break it into 2 lines in order to pass lint -- the auto-formatting was super buggy using `as Promise<..>` syntax